Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -630,6 +533,73 @@ private static void Clean(NugetPackageIdentifier package)
             }
         }
 
+        public static NugetFrameworkGroup GetBestDependencyFrameworkGroupForCurrentSettings(NugetPackage package)
+        {
+            var targetFrameworks = package.Dependencies
+                .Select(x => x.TargetFramework);
+
+            string bestTargetFramework = TryGetBestTargetFrameworkForCurrentSettings(targetFrameworks);
+            return package.Dependencies
+                .FirstOrDefault(x => x.TargetFramework == bestTargetFramework) ?? new NugetFrameworkGroup();
+        }
+
+        public static NugetFrameworkGroup GetBestDependencyFrameworkGroupForCurrentSettings(NuspecFile nuspec)
+        {
+            var targetFrameworks = nuspec.Dependencies
+                .Select(x => x.TargetFramework);
+
+            string bestTargetFramework = TryGetBestTargetFrameworkForCurrentSettings(targetFrameworks);
+            return nuspec.Dependencies
+                .FirstOrDefault(x => x.TargetFramework == bestTargetFramework) ?? new NugetFrameworkGroup();
+        }
+
+        private static readonly string[] unityFrameworks = new string[] { "unity" };
+        private static readonly string[] netStandardFrameworks = new string[] {
+            "netstandard2.0", "netstandard1.6", "netstandard1.5", "netstandard1.4", "netstandard1.3", "netstandard1.2", "netstandard1.1", "netstandard1.0" };
+        private static readonly string[] net4Frameworks = new string[] { "net462", "net461", "net46", "net452", "net451", "net45", "net403", "net40", "net4" };
+        private static readonly string[] net3Frameworks = new string[] { "net35-unity full v3.5", "net35-unity subset v3.5", "net35", "net20", "net11" };
+
+        public static string TryGetBestTargetFrameworkForCurrentSettings(IEnumerable<string> targetFrameworks)
+        {
+            int intDotNetVersion = (int)DotNetVersion; // c
+            //bool using46 = DotNetVersion == ApiCompatibilityLevel.NET_4_6; // NET_4_6 option was added in Unity 5.6
+            bool using46 = intDotNetVersion == 3; // NET_4_6 = 3 in Unity 5.6 and Unity 2017.1 - use the hard-coded int value to ensure it works in earlier versions of Unity
+            bool usingStandard2 = intDotNetVersion == 6; // using .net standard 2.0
+
+            var frameworkGroups = new List<string[]> { unityFrameworks };
+
+            if (usingStandard2) { frameworkGroups.Add(netStandardFrameworks); }
+            else if (using46) { frameworkGroups.Add(net4Frameworks); }
+            else { frameworkGroups.Add(net3Frameworks); }
```

Comment:
Dropping the net3x frameworks is a diversion from past behavior, and not correct.